services/rollout-service/pkg/cmd/server.go

```diff
@@ -109,7 +110,13 @@ func runServer(ctx context.Context, config Config) error {
 	}
 
 	var opts apiclient.ClientOptions
-	opts.ServerAddr = config.ArgocdServer
+	opts.ConfigPath = ""
+	u, err := url.ParseRequestURI(config.ArgocdServer)
+	if err != nil {
+		return fmt.Errorf("invalid argocd server url: %w", err)
+	}
+	opts.ServerAddr = u.Host
+	opts.PlainText = u.Scheme == "http"
 	opts.UserAgent = "kuberpult"
 	opts.Insecure = config.ArgocdInsecure
 	opts.AuthToken = config.ArgocdToken
```

Comment:
maybe extrac this as a function to make it better testable

Reply:
Done!